Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py

```diff
@@ -354,16 +355,41 @@ def create_pull_request(cls, repository, args, branch_point, callback=None, unbl
 
         existing_pr = None
         if remote_repo.pull_requests:
+            user, _ = remote_repo.credentials(required=False)
+
             log.info("Checking if PR already exists...")
             existing_pr = cls.find_existing_pull_request(repository, remote_repo)
             log.info("PR #{} found.".format(existing_pr.number) if existing_pr else "PR not found.")
-            if existing_pr and not existing_pr.opened and not args.defaults and (args.defaults is False or Terminal.choose(
-                    "'{}' is already associated with '{}', which is closed.\nWould you like to create a new pull-request?".format(
-                        repository.branch, existing_pr,
-                    ), default='No',
-            ) == 'Yes'):
+
+            if existing_pr and user and existing_pr.author != user and (args.defaults or Terminal.choose(
+                "'{}' is owned by '{}'\nWould you like to take over this pull-request?".format(existing_pr, existing_pr.author),
+                default='No',
+            ) == 'No'):
```

Comment:
UI nit: Could we better explain that if you _don't_ take over, it'll create a new PR for you? Something like:

    Take over this pull-request instead of creating a new one? (Yes/[No]):

I don't want people to think their only option is to take over the PR or cancel.

We could even switch from Y/N prompts to action verbs. Something like:

    You can either:
      [C]reate a new pull-request
      [O]verwrite {PR} and assign yourself
    How should git-webkit proceed? ([C]reate/[o]verwrite)